File: src/foreign_if/python/UT/src/gmm/test006.py
# Test means dtype
import sys
import numpy as np
from frovedis.exrpc.server import FrovedisServer
from frovedis.matrix.dense import FrovedisRowmajorMatrix
from frovedis.mllib.gmm import GaussianMixture


# initializing the Frovedis server
argvs = sys.argv
argc = len(argvs)
if (argc < 2):
    print ('Please give frovedis_server calling command as the first argument \n(e.g. "mpirun -np 2 -x /opt/nec/nosupport/frovedis/ve/bin/frovedis_server")')
    quit()
FrovedisServer.initialize(argvs[1])

train_mat = np.loadtxt("./input/gmm_data.txt")

# creating spectral agglomerative object
n_components = 2

try:
    gmm_model = GaussianMixture(n_components=n_components)
    # fitting the training matrix on gaussian mixture object
    gmm_model.fit(train_mat)
    cov = gmm_model.covariances_
except Exception as e:
    print ("status=Exception: " + str(e))
    sys.exit(1)
    

if(cov.dtype == np.float64):
    print("status=Passed")
else:
    print("status=Failed")

File: src/foreign_if/spark/UT/src/gmm/test_006.scala
package test.scala;

import com.nec.frovedis.Jexrpc.FrovedisServer
import com.nec.frovedis.mllib.clustering.GaussianMixture
import com.nec.frovedis.mllib.clustering.GaussianMixtureModel
import com.nec.frovedis.matrix.FrovedisRowmajorMatrix
import org.apache.spark.{SparkConf, SparkContext}
import org.apache.spark.mllib.linalg.Vectors

//OBJECTIVE: To test GaussianMixture for covariance dimensions

object GenericTest {

  def main(args: Array[String]): Unit = {

    // -------- configurations --------
    val conf = new SparkConf().setAppName("GenericTest").setMaster("local[2]")
    val sc = new SparkContext(conf)

    // initializing Frovedis server with "personalized command", if provided in command line
    if(args.length != 0) FrovedisServer.initialize(args(0))

    // -------- create Frovedis row major matrix--------
    val s_data = sc.textFile("./input/gmm/gmm_data.txt")
                   .map(s => Vectors.dense(s.split(' ').map(_.toDouble)))

    //------- Converting into Frovedis data
    val fdata = new FrovedisRowmajorMatrix(s_data)    

    // -------- training --------
    val ncomponents = 2
    val gmm = new GaussianMixture().setK(ncomponents)
                                   .setCovarianceType("full")  
                                   .setMaxIterations(100)
                                   .setConvergenceTol(0.01)
                                   .setInitParams("kmeans")
                                   .setSeed(1)
    val gmm_model = gmm.run(fdata)
    gmm_model.save("./out/gmm_model")
    
    val new_model = GaussianMixtureModel.load("./out/gmm_model")
     
    if(new_model.k == ncomponents) println("Status: Passed")
    else println("Status: Failed")    
    
    // -------- clean-up --------
    gmm_model.release() 
 
    FrovedisServer.shut_down()
    sc.stop()
  }
}
